modules/e2ee/Worker.js

```diff
@@ -82,6 +82,21 @@ const code = `
             this._id = id;
         }
 
+        /**
+         * Derives a per-participant key.
+         * @param {Uint8Array} keyBytes - Value to derive key from
+         * @param {Uint8Array} salt - Salt used in key derivation
+         */
+        async deriveKey(keyBytes, salt) {
+            const encoder = new TextEncoder();
+            const idBytes = encoder.encode(this._id);
+            // Separate both parts by a null byte to avoid ambiguity attacks.
+            const participantSalt = new Uint8Array(salt.byteLength + idBytes.byteLength + 1);
+            participantSalt.set(salt);
+            participantSalt.set(idBytes, salt.byteLength + 1);
```

Comment:
Maybe set the null byte explictly instead of relying on the array beeing zeroed by default?

Reply:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Uint8Array/Uint8Array -- nulling is in the spec

Reply:
👍